csrc/transform_replay.cpp

```diff
@@ -829,7 +830,8 @@ std::pair<TensorDomain*, size_t> TransformReplay::replayCasP(
           consumer->toString());
       new_allocation_domain.emplace_back(it->second);
     }
-    replayed->setAllocationDomain(std::move(new_allocation_domain), true);
+    replayed->setAllocationDomain(
+        std::move(new_allocation_domain), producer->getContiguity());
```

Comment:
I think @naoyam's concern makes sense. I think in the loop above, we should do something like:
```C++
    for (auto [id, contig] : producer->getAllocationDomainAndContiguity()) {
      auto it = p2c_map.find(id);
      NVF_CHECK(
          it != p2c_map.end(),
          "Unable to replayCasP: can not map ",
          id->toString(),
          " in the allocation domain of producer tensor ",
          producer->toString(),
          " to consumer tensor ",
          consumer->toString());
      new_allocation_domain.emplace_back(it->second);
      new_contiguity.emplace_back(contig);
    }
```

Reply:
Done.

Reply:
> I think @naoyam's concern makes sense. I think in the loop above, we should do something like:
> 
> ```c++
>     for (auto [id, contig] : producer->getAllocationDomainAndContiguity()) {
>       auto it = p2c_map.find(id);
>       NVF_CHECK(
>           it != p2c_map.end(),
>           "Unable to replayCasP: can not map ",
>           id->toString(),
>           " in the allocation domain of producer tensor ",
>           producer->toString(),
>           " to consumer tensor ",
>           consumer->toString());
>       new_allocation_domain.emplace_back(it->second);
>       new_contiguity.emplace_back(contig);
>     }
> ```

@zasdfgbnm Hmm, what about consumer broadcast domains, like the case as I mentioned above? `replayCasP` will ignore such consumer broadcast domains, so they should just remain in the leaf domain. It won't affect the allocation, so conceptually it should have no effect but mechanically aren't we assuming root broadcast domains remain in the allocation domain?

Reply:
Fortunately, `TransformReplay::replayCasP` is **currently** only used in `cacheBefore`, `cacheAfter` and `cacheFork`, so `consumer` is always a `set` of `producer`. The only tricky case at this moment is when `producer` is the product of a reduction, which I added a test for. 

However, your concern is valid. I actually don't know what the right behavior should be. As you said, the allocation order of consumer broadcast domains is ambiguous.

Reply:
Can you please add an assertion just in case?

Or, actually, I *think* we can just put a broadcast contiguity value in any position as it doesn't matter for the allocation. We would just need it for the consistency of the allocation domain.

Reply:
> Can you please add an assertion just in case?

Done.